### PR TITLE
Throw exception when projectID is missing from FIROptions

### PR DESCRIPTION
**Pull request by karenyz**
If `projectID` is missing the fetch request can't succeed, so we can fail the request before sending it to the server. This is causing some downstream issues (tracked in b/167621250)

### CODE REVIEW

File: FirebaseRemoteConfig/Sources/RCNConfigFetch.m
Comment (maksymmalyhin) [inline FirebaseRemoteConfig/Sources/RCNConfigFetch.m:222]: Why the conditions here and in FIRRemoteConfigComponent.m are different?

**Comment by paulb777** [comment]
FIS, an RC dependency, should already be checking the FIROptions, so it's not clear what this is changing. Also, this might be considered a breaking change, so assigning to @ryanwilson to approve from an API perspective.

**Review by maksymmalyhin — APPROVED** [review]
LGTM.

Note, the change could be considered as a breaking, but most like is not. RC has a dependency on FIS which already [throws on the missing project ID](https://github.com/firebase/firebase-ios-sdk/blob/1f2f5f617e6b964b8377d9f2dddf1d5225e9efe2/FirebaseInstallations/Source/Library/FIRInstallations.m#L130), so the exception won't be new for existing RC clients.

Please wait for @ryanwilson for the final approval on the API change.

**Comment by ryanwilson** [comment]
> LGTM.
> 
> Note, the change could be considered as a breaking, but most like is not. RC has a dependency on FIS which already [throws on the missing project ID](https://github.com/firebase/firebase-ios-sdk/blob/1f2f5f617e6b964b8377d9f2dddf1d5225e9efe2/FirebaseInstallations/Source/Library/FIRInstallations.m#L130), so the exception won't be new for existing RC clients.
> 
> Please wait for @ryanwilson for the final approval on the API change.

Looking at the internal bug though, it seems like they're getting through even with the dependency on FIS so I'm concerned this is going to break someone. Do we know if that's an older version sending the data, maybe pre-FIS? @karenyz

**Comment by maksymmalyhin** [comment]
> Looking at the internal bug though, it seems like they're getting through even with the dependency on FIS so I'm concerned this is going to break someone. Do we know if that's an older version sending the data, maybe pre-FIS? @karenyz

Note, FIS started throwing the exception on the missing project ID since last major version bump.

**Comment by karenyz** [comment]
I was hoping to fix-forward so we would stop sending invalid URLs moving forward, but since FIS is already throwing an exception it seems like this change doesn't make much of a difference (maybe makes it clearer for RC users?)

@ryanwilson yeah the malformed requests are from older versions, so this change wouldn't affect those requests

File: FirebaseRemoteConfig/Sources/RCNConfigFetch.m
Comment (erikeldridge) [inline FirebaseRemoteConfig/Sources/RCNConfigFetch.m:525]: I've recommended just asserting project ID as a required input in FIRRemoteConfigComponent.m. Given that, and since project ID is the only thing we're checking here, I think we can simplify this class by removing this logging.

File: FirebaseRemoteConfig/Tests/Unit/FIRRemoteConfigComponentTest.m
Comment (erikeldridge) [inline FirebaseRemoteConfig/Tests/Unit/FIRRemoteConfigComponentTest.m:170]: The fact we're only asserting FIRRemoteConfigComponentTest.m throws supports the idea that validating inputs there is sufficient.

**Review by erikeldridge — COMMENTED** [review]
We chatted in person, and this change may not be necessary, but if we do make it, I've added comments voting in favor of just asserting required configuration in FIRRemoteConfigComponent.m.

**Comment by ryanwilson** [comment]
LGTM from an API standpoint since this is a no-op

File: FirebaseRemoteConfig/Sources/RCNConfigFetch.m
Comment (karenyz) [reply FirebaseRemoteConfig/Sources/RCNConfigFetch.m]: removing the check here since we already check in `FIRRemoteConfigComponent`

**Review by erikeldridge — APPROVED** [review]
Shipit! 🚢

**Pull request merged**
